File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i in range(len(nums)):
            d[target - nums[i]] = i
        for i in range(len(nums)):
            if nums[i] in d and i != d[nums[i]]:
                return [d[nums[i]], i] if d[nums[i]] < i else [i, d[nums[i]]]

        

File: Data Structures & Algorithms/combination-sum-iv/submission-1.java
public class Solution {
    public int combinationSum4(int[] nums, int target) {
        Map<Integer, Integer> dp = new HashMap<>();
        dp.put(0, 1);

        for (int total = 1; total <= target; total++) {
            dp.put(total, 0);
            for (int num : nums) {
                dp.put(total, dp.get(total) + dp.getOrDefault(total - num, 0));
            }
        }
        return dp.get(target);
    }
}